optimum/neuron/distributed/base.py

```diff
@@ -293,10 +305,35 @@ def parallelize(
                 tied_weights[parameter] = new_parameter
                 new_parameters.add(new_parameter)
 
-            for mod in modules_to_initialize:
-                # This module has not pre-trained weights, it must be fine-tuned, we initialize it with the
-                # `reset_parameters()` method.
-                mod.reset_parameters()
+            for mod, parameter_names in modules_to_initialize.items():
+                if isinstance(mod, torch.nn.Embedding):
+                    # This module has not pre-trained weights, it must be fine-tuned, we initialize it with the
+                    # `reset_parameters()` method since there is only one parameter in torch.nn.Embedding.
+                    left_uninitialized = try_to_hf_initialize(model, mod, parameter_names)
+                    if left_uninitialized:
+                        mod.reset_parameters()
+                elif isinstance(mod, torch.nn.Linear):
+                    # This module has not pre-trained weights, it must be fine-tuned, we initialize it with the
+                    # `reset_parameters()` method but we need to be careful because one of the parameters might not
+                    # need initialization.
+                    left_uninitialized = try_to_hf_initialize(model, mod, parameter_names)
+                    if not left_uninitialized:
+                        continue
+                    initialize_linear(mod, left_uninitialized)
+
+                elif isinstance(mod, parallel_layers.layers.BaseParallelLinear):
+                    # First, we try to initialize the layer similarly as it would be done with the model.
+                    # To do that it is necessary to change the model class to that the `model._init_weights` method
+                    # considers this module as a `torch.nn.Linear` instance.
+                    orig_class = mod.__class__
+                    mod.__class__ = torch.nn.Linear
+                    left_uninitialized = try_to_hf_initialize(model, mod, parameter_names)
+                    mod.__class__ = orig_class
+                    if not left_uninitialized:
+                        continue
+                    initialize_parallel_linear(mod, left_uninitialized)
```

Comment:
I might miss something, but this seems a bit convoluted just to set the weights to a normal distribution and the bias to zero (which is exactly what the HF init does for all Linear layers).

Reply:
So we are not sure it is always the case, and it is not, for instance in T5. 
Basically it tries to use the initialization defined by the model if possible, and use the default initialization scheme otherwise.